### PR TITLE
Closes issue #6287 using optional chaining

### LINKED ISSUE #6287: Client side error upon blank JSON input in playground JSON textarea
## Bug creation:
1. Go to the link: [LinkFree Playground](https://linkfree.io/playground).
2. Add your username.
3. Add a blank JSON like this `{}`.
4. Click on the format JSON button.
5. Click on validate JSON button.
6. Click on the preview JSON button.

### Screenshots

https://user-images.githubusercontent.com/61611599/233778728-531d4448-c13e-48a2-8151-57bf33136a85.mov

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by omlondhe**
## Fixes Issue
Fixes #6287 

1. I followed the same steps mentioned in issue #6287 on the forked project locally.
2. I found the following error on my local playground page, the error was in the UserPage.js component, on line number 37.
Screenshot 2023-04-22 at 15 47 33

3. After solving this bug using the spread operator, I found the following bug in the UserLinks.js component, on line number 6.
Screenshot 2023-04-22 at 15 50 50

4. After following this bug, again using the spread operator the bug was found on the UserLinks.js component, on the next line, line number 7.
Screenshot 2023-04-22 at 15 51 42

5. Solved all the bugs and now the application works perfectly fine.
Screenshot 2023-04-23 at 10 57 30

## Changes proposed
1. An optional chaining is added to all of the above-mentioned lines and must be.
2. This is working as a check to determine if we have data to iterate through or if the dependent object is not undefined or null.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
Screenshots are added above for reference.

## Note to reviewers
The issue was assigned to me by @eddiejaoude and we discussed solving this by using some kind of basic checks, so I solved it using optional chaining. Up for any suggestions and corrections for this PR and up for solving other issues too.

Thank you!

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6304"></a>

### CODE REVIEW

**Comment by omlondhe** [comment]
Hey @SaraJaoude , how much time will it take for reviewers to close this PR? I want to contribute to more issues after this PR is closed. 

Thank you!

**Comment by SaraJaoude** [comment]
> Hey @SaraJaoude , how much time will it take for reviewers to close this PR? I want to contribute to more issues after this PR is closed.
> 
> Thank you!

@omlondhe as mentioned in our Contributing Guide which you have been sent before, please avoid tagging a specific maintainer.  All maintainers can reply to these questions and receive notifications.

We appreciate your enthusiasm to want to work on other issues.  However please remember we are all working on this voluntarily, have other commitments (both professional and personal) and are located in different timezones.  If you take a look at the Issue list, you will also see that there are items which are marked as 'critical' so these will take priority. Therefore please be patient - someone will look at this issue when they next have a chance.

**Comment by omlondhe** [comment]
Got it, I apologise for tagging you, completely forgot about the rules, will not tag anyone.
Sorry!

File: components/user/UserLinks.js [hunk 1]
```diff
@@ -3,8 +3,10 @@ import Alert from "@components/Alert";
 
 export default function UserLinks({ BASE_URL, data }) {
   const defaultBucket = "Others";
-  data.links = data.links.map((link, i) => ({ id: i, ...link }));
-  const buckets = data.links.reduce((acc, obj) => {
+  // Solved bug 2.1 shown in the image
```
Comment (eddiejaoude) [inline components/user/UserLinks.js:6]: what is `bug 2.1`?

File: components/user/UserPage.js [hunk 2]
```diff
@@ -33,8 +33,8 @@ export default function UserPage({ data, BASE_URL }) {
       }
       return [];
     }
-
-    return { ...tab, total: data.links.length };
+    // solved bug 1 shown in image
```
Comment (eddiejaoude) [inline components/user/UserPage.js:36]: what does `bug 1` mean?

File: components/user/UserLinks.js (hunk 1 above)
Comment (omlondhe) [reply components/user/UserLinks.js:6]: Bug 2.1 and 2.2 refers to the following screenshots screenshot, the second and third screenshots from the PR respectively:

Screenshot 2023-04-22 at 15 50 50
This is 2.1

<br/><br/>
Screenshot 2023-04-22 at 15 51 42
This is 2.2

Comment (omlondhe) [reply components/user/UserLinks.js:6]: Removing them

Comment (omlondhe) [reply components/user/UserLinks.js]: Removed all those comments, can you check on your side?

**Review by eddiejaoude — APPROVED** [review]
Looks good, thank you 👍

**Pull request merged**